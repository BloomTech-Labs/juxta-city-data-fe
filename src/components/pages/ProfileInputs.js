import React from "react";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";
import SelectState from "./SelectState";
import TextField from "@material-ui/core/TextField";
import formStyles from "./profileStyles/formStyles";

function ProfileInputs({ handleSubmit, addHandleChange }) {
  const classes = formStyles();

  return (
    <Grid item xs={6}>
      <Paper className={classes.paper}>
        <form onSubmit={handleSubmit} className={classes.root}
          noValidate autoComplete="off">
          <TextField label="First Name" type="text" name="first_name" onChange={addHandleChange} />
          <TextField label="Last Name" type="text" name="last_name" onChange={addHandleChange} />
          <TextField label="Birthdate"type="date"name="dob" onChange={addHandleChange}
            InputLabelProps={{shrink: true}} />
          <TextField label="Address"type="text"name="address" onChange={addHandleChange} />
          <TextField label="City" type="text" name="city" onChange={addHandleChange} />
          <SelectState addHandleChange={addHandleChange} />
          <TextField label="Zip code" type="number" name="zip" onChange={addHandleChange} />
          <Button type="submit" variant="outlined" color="primary" className={classes.button}>Add profile</Button>
        </form>
      </Paper>
    </Grid>
  );
}
export default ProfileInputs;
