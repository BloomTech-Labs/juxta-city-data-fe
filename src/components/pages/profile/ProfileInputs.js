import React from "react";
import "../../../index.scss";
import StateSelect from "./StateSelect";
import ZipCode from "./ZipCode";
import {
  handleSubmitFocus,
  handleTextFocus
} from "./SVGfunctions"; 
import SVGstyle from './SVGstyle'

function ProfileInputs({ handleSubmit, addHandleChange }) {

  const inputData = [ {name: "first_name", placeholder: "First Name", type: "text", "dataStrokedashoffset":"0"},
  { name: "last_name", placeholder: "Last Name", type: "text", "dataStrokedashoffset":"-331" },
  { name: "dob", placeholder: "Date of birth", type: "date",  "dataStrokedashoffset":"-686"},
  { name: "address", placeholder: "Address", type: "text", "dataStrokedashoffset": "-1012"},
  { name: "city", placeholder: "City", type: "text",  "dataStrokedashoffset":"-1367"}];

 
  return (
    <div className="demensions">
      <div>
        <SVGstyle />
        <form
          style={{ position: "absolute", zIndex: 10, margin: "0 40px" }}
          noValidate
          autoComplete="off"
          onSubmit={handleSubmit}
        >
          {inputData.map((item, index) => (
            <div key={index}>
              <label className="form-label" htmlFor={item.name}>
                {item.placeholder}
              </label>
              <input
                data-testid={item.name}
                name={item.name}
                type={item.type}
                placeholder={item.placeholder}
                className="input"
                data-strokedashoffset={item.dataStrokedashoffset}
                data-strokedasharray="240 3000"
                onFocus={handleTextFocus}
                onChange={addHandleChange}
                aria-label={item.placeholder}
              />
            </div>
          ))}
          <StateSelect
            addHandleChange={addHandleChange}
            handleTextFocus={handleTextFocus}
          />
          <ZipCode
            addHandleChange={addHandleChange}
            handleTextFocus={handleTextFocus}
          />

          <div className="center">
            <button type="submit" onFocus={handleSubmitFocus} className="btn" data-testid="submit">
              Add Profile
            </button>
          </div>
        </form>
      </div>
    </div>
  );
}

export default ProfileInputs;
