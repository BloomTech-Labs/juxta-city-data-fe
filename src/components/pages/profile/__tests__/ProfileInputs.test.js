import React from 'react';
import ProfileInputs from '../ProfileInputs';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer'
// import '@testing-libarary/jest-dom/extend-expect';
import Adapter from 'enzyme-adapter-react-16';
// import { render, fireEvent, cleanup } from '@testing-library/react';

configure({ adapter: new Adapter() })

// afterEach(cleanup)

//testing a controlled component form.

describe('the profile input ', () => {
    const wrapper = shallow(<ProfileInputs />)

    it('snap shot the component', () => {
        const shot = renderer.create(<ProfileInputs />).toJSON()
        expect(shot).toMatchSnapshot()
    })

    it('checks the length of all the content in the profileInputs', () => {
        expect(wrapper.find('form')).toHaveLength(1)
        expect(wrapper.find('svg')).toHaveLength(1)
    })

    it('checks the ')
})