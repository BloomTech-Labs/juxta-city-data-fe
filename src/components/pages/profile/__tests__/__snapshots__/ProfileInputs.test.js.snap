// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the profile input  snap shot the component 1`] = `
<div
  className="demensions"
>
  <div>
    <svg
      className="svg"
      viewBox="0 0 320 550.8"
    >
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="linearGradient"
          inkscapeCollect="always"
          x1="13"
          x2="307"
          y1="193.49992"
          y2="193.49992"
        >
          <stop
            id="stop876"
            offset="0"
            style={
              Object {
                "stopColor": "#ff00ff",
              }
            }
          />
          <stop
            id="stop878"
            offset="1"
            style={
              Object {
                "stopColor": "#ff0000",
              }
            }
          />
        </linearGradient>
      </defs>
      <path
        className="path"
        d="M282.8,23l-240,0c0,0-25,0.8-25,35c0,34.2,25,35,25,35l-5.5,0l240,0c0,0,25,0.8,25,35c0,34.2-25,35-25,35l5.5,0l-240,0 c0,0-25,0.8-25,35c0,34.2,25,35,25,35l-5.5,0l240,0c0,0,25,0.8,25,35c0,34.2-25,35-25,35l8.3,0l-240,0c0,0-25,0.8-25,35 c0,34.2,25,35,25,35l-5.5,0l240,0c0,0,25,0.8,25,35c0,34.2-25,35-25,35H40c0,0-25,4-25,38.5S40,520,40,520h215c0,0,20-1,20-25 s-20-25-20-25H65c0,0-20,1.7-20,25c0,24,20,25,20,25h168.6"
      />
    </svg>
    <form
      autoComplete="off"
      noValidate={true}
      style={
        Object {
          "margin": "0 40px",
          "position": "absolute",
          "zIndex": 10,
        }
      }
    >
      <label
        alt="first name"
        aria-label="first_name"
        className="label"
        htmlFor="first_name"
      >
        First Name
      </label>
      <input
        class="input"
        data-strokedasharray="240 3000"
        data-strokedashoffset="0"
        name="first_name"
        onFocus={[Function]}
        placeholder="First Name"
        type="text"
      />
      <label
        aria-label="last name"
        className="label"
        data-testid="lname"
        htmlFor="last_name"
      >
        Last Name
      </label>
      <input
        aria-label="last name"
        class="input"
        data-strokedasharray="240 3000"
        data-strokedashoffset="-331"
        name="last_name"
        onFocus={[Function]}
        placeholder="Last Name"
        type="text"
      />
      <label
        aria-label="birth date"
        className="label"
        htmlFor="dob"
      >
        Date
      </label>
      <input
        class="input datepicker"
        data-strokedasharray="240 3000"
        data-strokedashoffset="-686"
        id="dob"
        name="dob"
        onFocus={[Function]}
        type="date"
      />
      <label
        aria-label="address"
        className="label"
        htmlFor="address"
      >
        Address
      </label>
      <input
        className="input"
        data-strokedasharray="240 3000"
        data-strokedashoffset="-1012"
        name="address"
        onFocus={[Function]}
        placeholder="Address"
        type="text"
      />
      <label
        aria-label="city"
        className="label"
        htmlFor="city"
      >
        City 
      </label>
      <input
        className="input"
        data-strokedasharray="240 3000"
        data-strokedashoffset="-1367"
        name="city"
        onFocus={[Function]}
        placeholder="City"
        type="text"
      />
      <div>
        <label
          aria-label="state"
          className="label"
        >
          States
        </label>
        <select
          className="input"
          data-strokedasharray="240 3000"
          data-strokedashoffset="-1696"
          name="state"
          onFocus={[Function]}
        >
          <option
            placeholder="Choose"
          >
            Choose...
          </option>
          <option
            value="Alabama"
          >
            Alabama
          </option>
          <option
            value="Alaska"
          >
            Alaska
          </option>
          <option
            value="Arizona"
          >
            Arizona
          </option>
          <option
            value="Arkansas"
          >
            Arkansas
          </option>
          <option
            value="California"
          >
            California
          </option>
          <option
            value="Colorado"
          >
            Colorado
          </option>
          <option
            value="Connecticut"
          >
            Connecticut
          </option>
          <option
            value="Delaware"
          >
            Delaware
          </option>
          <option
            value="Florida"
          >
            Florida
          </option>
          <option
            value="Georgia"
          >
            Georgia
          </option>
          <option
            value="Hawaii"
          >
            Hawaii
          </option>
          <option
            value="Idaho"
          >
            Idaho
          </option>
          <option
            value="Illinois"
          >
            Illinois
          </option>
          <option
            value="Indiana"
          >
            Indiana
          </option>
          <option
            value="Iowa"
          >
            Iowa
          </option>
          <option
            value="Kansas"
          >
            Kansas
          </option>
          <option
            value="Kentucky"
          >
            Kentucky
          </option>
          <option
            value="Louisiana"
          >
            Louisiana
          </option>
          <option
            value="Maine"
          >
            Maine
          </option>
          <option
            value="Maryland"
          >
            Maryland
          </option>
          <option
            value="Massachusetts"
          >
            Massachusetts
          </option>
          <option
            value="Michigan"
          >
            Michigan
          </option>
          <option
            value="Minnesota"
          >
            Minnesota
          </option>
          <option
            value="Mississippi"
          >
            Mississippi
          </option>
          <option
            value="Missouri"
          >
            Missouri
          </option>
          <option
            value="Montana"
          >
            Montana
          </option>
          <option
            value="Nebraska"
          >
            Nebraska
          </option>
          <option
            value="Nevada"
          >
            Nevada
          </option>
          <option
            value="New Hampshire"
          >
            New Hampshire
          </option>
          <option
            value="New Jersey"
          >
            New Jersey
          </option>
          <option
            value="New Mexico"
          >
            New Mexico
          </option>
          <option
            value="New York"
          >
            New York
          </option>
          <option
            value="North Carolina"
          >
            North Carolina
          </option>
          <option
            value="North Dakota"
          >
            North Dakota
          </option>
          <option
            value="Ohio"
          >
            Ohio
          </option>
          <option
            value="Oklahoma"
          >
            Oklahoma
          </option>
          <option
            value="Oregon"
          >
            Oregon
          </option>
          <option
            value="Pennsylvania"
          >
            Pennsylvania
          </option>
          <option
            value="Rhode Island"
          >
            Rhode Island
          </option>
          <option
            value="South Carolina"
          >
            South Carolina
          </option>
          <option
            value="South Dakota"
          >
            South Dakota
          </option>
          <option
            value="Tennessee"
          >
            Tennessee
          </option>
          <option
            value="Texas"
          >
            Texas
          </option>
          <option
            value="Utah"
          >
            Utah
          </option>
          <option
            value="Vermont"
          >
            Vermont
          </option>
          <option
            value="Virginia"
          >
            Virginia
          </option>
          <option
            value="Washingtion"
          >
            Washington
          </option>
          <option
            value="West Virginia"
          >
            West Virginia
          </option>
          <option
            value="Wisconsin"
          >
            Wisconsin
          </option>
          <option
            value="Wyoming"
          >
            Wyoming
          </option>
        </select>
      </div>
      <div>
        <label
          aria-label="zip code"
          className="label"
          for="Zip Code"
        >
          Zip
        </label>
        <input
          className="input"
          data-strokedasharray="240 3000"
          data-strokedashoffset="-2040"
          name="zip"
          onFocus={[Function]}
          placeholder="Zip Code"
          type="number"
        />
      </div>
      <div
        className="center"
      >
        <button
          className="btn"
          onFocus={[Function]}
          type="submit"
        >
          Add Profile
        </button>
      </div>
    </form>
  </div>
</div>
`;
